Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -155,34 +152,20 @@ void FindSXPeaks::exec() {
     double rightBinEdge = *std::next(leftBinPosition);
     double tof = 0.5 * (leftBinEdge + rightBinEdge);
 
-    Geometry::IDetector_const_sptr det;
-    try {
-      det = localworkspace->getDetector(static_cast<size_t>(i));
-    } catch (Mantid::Kernel::Exception::NotFoundError &) {
-      // Catch if no detector. Next line tests whether this happened - test
-      // placed
-      // outside here because Mac Intel compiler doesn't like 'continue' in a
-      // catch
-      // in an openmp block.
-    }
     // If no detector found, skip onto the next spectrum
-    if (!det)
+    if (!spectrumInfo.hasDetectors(static_cast<size_t>(i))) {
       continue;
+    }
+    const auto &det = spectrumInfo.detector(static_cast<size_t>(i));
 
-    double phi = det->getPhi();
+    double phi = det.getPhi();
     if (phi < 0) {
       phi += 2.0 * M_PI;
     }
 
-    double th2 = det->getTwoTheta(sample, L1);
-
     std::vector<int> specs(1, i);
 
-    Mantid::Kernel::V3D L2 = det->getPos();
-    L2 -= sample;
-
-    SXPeak peak(tof, th2, phi, *maxY, specs, l1 + L2.norm(), det->getID(),
-                localworkspace->getInstrument());
```

Comment:
This is a weird: If the spectrum in the workspace has grouped detectors, `localworkspace->getDetector(static_cast<size_t>(i));` above returns a `DetectorGroup`. `det->getID()` will then return the ID of the first detector in the group, so `SXPeak` would use the position of only that first detector! All the other parameters that are passed to the constructor here, such as `th2` are for the whole group!

So in conclusion, I think the way you changed `SXPeak` (to use `SpectrumInfo`) makes sense, since actually there is a bug here (please double check yourself, maybe I am overlooking something!)? Maybe you can open an issue for this with a short description and mark it as fixed by this PR?